x-pack/platform/plugins/shared/onechat/server/services/tools/esql/storage.ts

```diff
@@ -0,0 +1,69 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import type { Logger, ElasticsearchClient } from '@kbn/core/server';
+import { ToolDescriptorMeta } from '@kbn/onechat-common';
+import { IndexStorageSettings, StorageIndexAdapter, types } from '@kbn/storage-adapter';
+
+export const esqlToolIndexName = '.kibana_onechat_esql_tools';
+
+const storageSettings = {
+  name: esqlToolIndexName,
+  schema: {
+    properties: {
+      id: types.keyword({}),
+      name: types.text({}),
+      description: types.text({}),
+      query: types.text({}),
+      params: types.object({
+        dynamic: true,
+        properties: {},
+      }),
+      meta: types.object({
+        properties: {
+          providerId: types.keyword({}),
```

Comment:
Thanks @seanstory for adding me as a reviewer here! I genuinely appreciate seeing the code early, but I'm not sure how to best be helpful here. For example, this should likely be `provider_id` for consistency with all other Kibana and Elasticsearch APIs, but I'm not sure this pull request is the best place to raise such nits.

Reply:
Whether you raise nits here, in slack, or in the design doc, matters less to me than being sure you have the context and the opportunity to raise concerns early. If you don't have any strong concerns, that's fine too. :)

Reply:
So for what it's worth, the storage schema and API schema can (and absolutely should) be dissociated. We should be able to have different field names for both of them (and even more structural differences).

Persistence schema !== API schema (even if we can start by having the two very similar, they **must** be able to diverge/differ).

So I think the right file to check would be `x-pack/platform/plugins/shared/onechat/server/routes/esql_tools.ts` 🙂